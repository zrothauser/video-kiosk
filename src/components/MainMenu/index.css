@keyframes pulse {
	0% { opacity: 0.5; }
	6% { opacity: 1; }
	20% { opacity: 1; }
	26% { opacity: 0.5; }
	100% { opacity: 0.5; }
}

.b-main-menu {
  height: 100vh;
}

.b-main-menu__list {
  display: flex;
  flex-direction: column;
  height: 100vh;
  margin: 0;
  padding: 0;
  list-style: none;
  z-index: 5;
}

.b-main-menu__item {
  margin: 0;
  padding: 0;
  flex: 1;
  display: flex;
  justify-content: center;
  flex-direction: column;
  z-index: 5;
  border-bottom: 1px solid rgba(255,255,255,0.5);
}

.b-main-menu__item:last-of-type {
  border-bottom: none;
}

.b-main-menu__link {
  display: block;
  margin-left: 75%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-left: 1.5951041667vw;
}

.b-main-menu__item .b-main-menu__link {
  opacity: 0.5;
}
