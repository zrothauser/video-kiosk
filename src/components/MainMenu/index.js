// Dependencies
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

// Styles
import './index.css';

const MainMenu = (props) => {
  const {
    categories,
    setSlug,
  } = props;

  return (
    <div className="b-main-menu">
      <span className="h-screen-reader">
        Main Menu
      </span>
      <ul className="b-main-menu__list">
        {categories.map((category, index) => (
          <li
            className="b-main-menu__item"
            key={category.slug}
          >
            <Link
              to={`/${setSlug}/${category.slug}`}
              className="b-main-menu__link"
              style={{
                animation: `pulse ${categories.length * 3}s infinite`,
                animationDelay: `${index * 3}s`,
              }}
            >
              {category.title}
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
};

MainMenu.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.object).isRequired,
  setSlug: PropTypes.string.isRequired,
};

export default MainMenu;
