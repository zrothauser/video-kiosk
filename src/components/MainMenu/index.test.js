import React from 'react';
import { shallow } from 'enzyme';
import MainMenu from './index';

describe('components', () => {
  describe('MainMenu', () => {
    it('renders without crashing', () => {
      const mockProps = {
        isLoading: false,
        categories: [],
      };

      shallow(<MainMenu {...mockProps} />);
    });
  });
});
