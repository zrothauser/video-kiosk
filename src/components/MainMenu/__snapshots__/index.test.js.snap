// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components MainMenu renders correctly 1`] = `
<div
  className="b-main-menu"
>
  <span
    className="h-screen-reader"
  >
    Main Menu
  </span>
  <ul
    className="b-main-menu__list"
  />
</div>
`;
